apps/rush-lib/src/utilities/AsyncRecycler.ts

```diff
@@ -73,7 +73,25 @@ export class AsyncRecycler {
     }
 
     Utilities.retryUntilTimeout(
-      () => FileSystem.move({ sourcePath: folderPath, destinationPath: newFolderPath }),
+      () => {
+        try {
+          FileSystem.move({ sourcePath: folderPath, destinationPath: newFolderPath });
+        } catch (e) {
+          if (FileSystem.isErrnoException(e)) {
+            // It could be an invalid symlink, so try to obtain the stats
+            try {
+              const stats: fs.Stats = FileSystem.getLinkStatistics(folderPath);
+              if (stats.isSymbolicLink() && stats.isDirectory()) {
+                FileSystem.deleteFolder(folderPath);
```

Comment:
Deleting folders can be a very long-running operation.  The `moveFolder()` API is expected to complete very fast, and then `deleteAll()` performs the time-consuming operation in a separate process.  The code here undoes those timing guarantees.

Is this related to PNPM workspaces somehow?  It would be useful to have a description of what problem we're trying to solve, and how it fits with the purpose of `AsyncRecycler.moveFolder()`.

Reply:
See above. I'm removing this functionality